### ISSUE TITLE
Compatibility with React version > 15.5

### ISSUE THREAD

**Issue by RickEyre** [styled-components/styled-components#691]
I recently upgraded to React 15.5. They've deprecated the usage of PropTypes from the React module.

Is there any plans to upgrade to the `prop-types` module?

Would you accept a pull request to fix this?

**Comment by RickEyre** [comment]
Sorry, just took a look through the change log.

**Comment by RickEyre** [comment]
Actually, I'm still getting an error when using version `2.0.0-15` and React > `0.15`

```bash
console.error node_modules/fbjs/lib/warning.js:36
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
```

Is this expected? If so, when should we expect a new release to deal with this? I'm opening to helping if needed.

**Comment by k15a** [comment]
AFAIK it's not released yet. /cc @mxstbr

**Comment by YasserKaddour** [comment]
This should be fixed for `styled-components` in `2.0.0-15`, check the stack trace to see what library caused the warning.

**Comment by RickEyre** [comment]
@YasserKaddour Yes, I see. It's coming from `fbjs`. I will open a ticket with them.
